crypto/bio/bio_sock.c

```diff
@@ -400,7 +400,7 @@ int BIO_socket_wait(int fd, int for_read, time_t max_time)
         return 1;
 
     now = time(NULL);
-    if (max_time <= now)
+    if (max_time < now)
```

Comment:
Why is this change necessary?

Reply:
Yes, I need something to get writability of a socket without waiting the full second.
this just allows the return value to be 1 in this case, I consider it a bug that this
function did always return 0 when the due time is reached, even if the socket
is writable now.

Reply:
Ah - right. Makes sense.